okhttp-client-handler/src/main/scala/com/softwaremill/sttp/okhttp/OkHttpClientHandler.scala

```diff
@@ -139,7 +149,7 @@ abstract class OkHttpAsyncClientHandler[R[_], S](client: OkHttpClient,
 
           override def onResponse(call: Call, response: OkHttpResponse): Unit =
             try success(readResponse(response, r.responseAs))
-            catch { case e: Exception => error(e) }
+            catch { case e: Exception => error(e) } finally response.close()
```

Comment:
Indeed, the responses need closing, however I think that's too aggressive - it fails in case of streams, as the response is closed before the stream is consumed (as the failing test shows).